src/types.ts

```diff
@@ -178,6 +180,7 @@ export interface HttpSettings {
   proxyConnectHeaders?: string[];
 
   // Authentication
+  authType?: HttpAuthType;
```

Comment:
Great feedback! I changed the implementation to use refs instead of polluting these type definitions with something that's just frontend related as you suggested 👍